### PR TITLE
Adding Tilt to Window Covering

### PR DESCRIPTION
**Pull request by giase82**
Hey!

Please have a look. Basis for this is "GiraJalousieActuator.js".

I have also removed the timer as this didn't work for me. The "onKNXValueChange" for "CurrentPosition" takes care about this once the shutter is at it's end-end-position.

This is tested in my environment and works afaik.

//giase

### CODE REVIEW

File: lib/addins/WindowCoveringTilt.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+					break;
+
+			} // switch
+
+		} else if (field === "TargetHorizontalTiltAngle") {
+			// TargetPosition is DPT5.001 Percentage (0..255)
+			// need to convert to (0..100) first
+			// Homekit is using %-open, meaning 0% is closed/down
+			
+			newValue = 100 - knxValue*100 / 255;
```
Comment (snowdd1) [inline lib/addins/WindowCoveringTilt.js]: According to Docu the HomeKit value is supposed [to be an angle from -90° to 90°](https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L1832)  
Can you make sure that it is *not* an angle but a percentage?  
Same for Current- and of course reverse conversion from HomeKit to KNX

**Comment by snowdd1** [comment]
Can you check the type conversions, please?

**Comment by snowdd1** [comment]
Hi @giase82 you might be interested in https://github.com/snowdd1/homebridge-knx/commit/786269f47ca2f4a934e4e3a96ab9b8757ca43983 :

https://github.com/snowdd1/homebridge-knx/blob/master/handler-add-in.md#thismyapigetpropertyfieldproperty

**Comment by giase82** [comment]
Cool Danke! I think I won't need it for now, but that will be very helpful in the future.
For now (see review) I created a local constant where you can specify (true/false) if you have a 90° shutter or not (meaning you have a 180°). The slider is untouched.
If you have configured it to be a 90° shutter only (as mine), the shutter will not react when moving the slider between -90 to 0.

**Pull request merged**